server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -739,11 +748,13 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder, final
                 builder.endObject();
             }
             final IndexVersion version = snapshotDetails.getVersion();
-            if (version != null) {
-                if (version.before(IndexVersion.V_8_9_0)) {
-                    builder.field(VERSION, Version.fromId(version.id()).toString());
+            if (version != null && version.id() != NUMERIC_INDEX_VERSION_MARKER.id()) {
```

Comment:
Would we not want to retain the version field if it is 8.11.0? I think this nulls it out, which would trigger an old version to read the details out I think.

Definitely a minor point (would require 8.11+ writing data, then 8.9 roundtrip, then 8.11+ roundtrip and finally 8.9 read to be a problem), but seems slightly better anyway. On the other hand, I am sure there is a good reason for this that I have not realized?

Reply:
Yes in fact this does not especially matter since we will re-read the `SnapshotInfo` either way. I added this to help with forensics a little: if we encounter a `RepositoryData` in the wild with `"version":"8.11.0"` and no `"index_version"` field alongside it then we know that it must have been written by a version before 8.11.0.

Reply:
Ok. So IIUC, 8.9 will parse the snapshot info's version as something like 8.50.00 and be happy about it.

Would it then be simpler to always null out the `version` field always (in 8.11+) and only write the `index_version` field? Any prior version would then read the details from snapshot info.

Reply:
I was wondering the exact same thing (why we need `version.id() != NUMERIC_INDEX_VERSION_MARKER.id()`). 
So 8.9 should keep "8.11" as a string version, but will clear out the numeric id out in the roundtrip; then 8.11 will read this (OK). When it's time for 8.11 to write, this `if` should not be passing, so no version or index_version will be written out.
But I assume you are saying this does not matter, because version will have be updated before writing (will not be 8.11 anymore, but something like 8_500_003)?

Reply:
If the version is null when reading, we [read it from the snapshot info](https://github.com/elastic/elasticsearch/blob/1c86da0fa4119d32f5eaaa61fe26291335dd5175/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java#L2362), also in prior versions. SnapshotInfo uses (and used) and int as xcontent serialization so a new version 8_500_003 would be parsed as 8.50.03 by some old ES version and thus be parseable (but ofc not restorable).